src/test/java/com/okta/tools/aws/settings/ConfigurationTest.java

```diff
@@ -105,28 +89,54 @@ void addOrUpdateProfileToNewConfigFile() throws IOException {
-        assertEquals(2, initiallyEmpty.settings.size());
-        assertEquals(profileName + "_source", initiallyEmpty.settings.get(DEFAULTPROFILENAME, SOURCE_PROFILE));
-        assertEquals(role_arn, initiallyEmpty.settings.get(DEFAULTPROFILENAME, ROLE_ARN));
+        assertEquals(2, initiallyEmpty.getSections().size());
+        assertEquals(profileName + "_source", initiallyEmpty.getProperty(DEFAULT_PROFILE_NAME, SOURCE_PROFILE));
+        assertEquals(role_arn, initiallyEmpty.getProperty(DEFAULT_PROFILE_NAME, ROLE_ARN));
         // State of the default profile after creating an initial profile.
-        final Map<String, String> defaultProfileBefore = sectionToMap.apply(initiallyEmpty.settings.get(DEFAULTPROFILENAME));
+        final Map<String, Object> defaultProfileBefore = initiallyEmpty.sectionToMap(DEFAULT_PROFILE_NAME);
 
         // Write another profile. Make sure the default profile is left alone.
         final String postfix = "_2";
         initiallyEmpty.addOrUpdateProfile(profileName + postfix, role_arn + postfix, region);
-        assertTrue(initiallyEmpty.settings.containsKey("custom " + profileName + postfix));
-        assertEquals(3, initiallyEmpty.settings.size());
-        assertEquals(defaultProfileBefore, sectionToMap.apply(initiallyEmpty.settings.get(DEFAULTPROFILENAME)));
+        assertTrue(initiallyEmpty.getSections().contains("custom " + profileName + postfix));
+        assertEquals(3, initiallyEmpty.getSections().size());
+        assertEquals(defaultProfileBefore, initiallyEmpty.sectionToMap(DEFAULT_PROFILE_NAME));
+    }
+
+    /*
+     * Test updating a profile with nested configuration (Issue #141).
+     */
+    @Test
+    void addOrUpdateProfileToExistingProfileWithNestedConfiguration() throws IOException {
+        final String existingCombined = manualRole + "\n"
+                + "s3 =\n"
+                + "    max_queue_size = 1000" + "\n\n" + existingProfile;
+
+        final StringReader configurationReader = new StringReader(existingCombined);
+        final StringWriter configurationWriter = new StringWriter();
+        final Configuration configuration = new Configuration(configurationReader);
+
+        final String updatedPrefix = "updated_";
+        final String expected = "[profile " + profileName + "]\n"
+                + Configuration.ROLE_ARN + " = " + updatedPrefix + role_arn + "\n"
+                + SOURCE_PROFILE + " = " + profileName + "_source\n"
+                + Configuration.REGION + " = " + region + "\n"
+                + "s3 = \n" // additional space here is a tolerable divergence (AWS CLI doesn't care)
+                + "    max_queue_size = 1000"
+                + "\n\n" + existingProfile;
+
+
+        configuration.addOrUpdateProfile(profileName, updatedPrefix + role_arn, region);
+        configuration.save(configurationWriter);
+
+        String given = StringUtils.remove(configurationWriter.toString().trim(), '\r');
+
+        assertEquals(expected, given);
```

Comment:
✅ worth adding a test to cover this nested config stuff seeing as it's what we came here to build.